ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/TransactionImporterImple.java

```diff
@@ -122,6 +112,14 @@ public TransactionImple recoverTransaction(Uid actId)
 		if (recovered.baseXid() == null)
 		    throw new IllegalArgumentException();
 		
+		return addImportedTransaction(recovered);
+	}
+
+	private TransactionImple addImportedTransaction(TransactionImple importedTransaction) throws XAException {
+		SubordinateXidImple importedXid = new SubordinateXidImple(importedTransaction.baseXid());
+		TransactionImpleHolder holder = new TransactionImpleHolder();
+		TransactionImpleHolder prevHolder = _transactions.putIfAbsent(importedXid, holder);
+
```

Comment:
Interruption doesn't actually affect the program flow; it only comes into play once something either checks the flag manually or uses a construct (like Object.wait()) which is directly sensitive to interruption.

I can see no statements between the add and the set in the finally block which can trigger interruption in any way that is detectable before the setter is called.  The risk point would probably be line 140, but in this case the caller isn't the setting thread so again it doesn't influence the final set operation as far as I can see.